BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -1123,6 +1175,290 @@ void Visualize()
+                                priceCkeckoout += (kvp.Value.Count * pb); //and tally total budget spent so far
+                            }
+                            else
+                            {
+                                notOnPriceList = true;
+                            }
+                        }
+                        foreach (var partModule in kvp.Value[0].Modules) //weapon whitelist/engine count
+                        {
+                            if (partModule == null) continue;
+                            switch (partModule.moduleName)
+                            {
+                                case "ModuleEngines":
+                                case "ModuleEnginesFX":
+                                    {
+                                        if (engineparts.Contains(kvp.Value[0].partInfo.title)) break; //don't grab both moduleEngines for dual-mode engines and double-count them
+                                        if (CompSettings.CompVesselChecksEnabled && maxEngines < maxPartCount || maxTWR > 0)
+                                        {
+                                            if (maxEngines < maxPartCount)
+                                            {
+                                                if (!string.IsNullOrEmpty(engineparts)) engineparts += " | ";
+                                                engineparts += $"{kvp.Value.Count}x {kvp.Value[0].partInfo.title}";
+                                                engineCount += kvp.Value.Count;
+                                                Debug.Log($"[VesselCheckDebug] found {kvp.Value.Count} {kvp.Value[0].partInfo.title}");
+                                            }
+                                            if (maxTWR > 0) maxThrust += kvp.Value[0].FindModuleImplementing<ModuleEngines>().maxThrust * kvp.Value.Count;
+                                        }
+                                        break;
+                                    }
+                                case "ModuleWeapon":
+                                case "MissileBase":
+                                case "MissileLauncher":
+                                    {
+                                        if (pointBuyBudget > 0 && notOnPriceList) //if a weapon isn't on the price list, it's banned
+                                        {
+                                            if (!string.IsNullOrEmpty(blacklistedParts)) blacklistedParts += " | ";
+                                            blacklistedParts += $"{kvp.Value[0].partInfo.title}({kvp.Value.Count}/0)";
+                                        }
+                                        break;
+                                    }
+                                case "MissileFire":
+                                    {
+                                        weaponmanagers++;
```

Comment:
This should be using `weaponmanagers += kvp.Value.Count`, otherwise it's only seeing the first one.
Similarly for the AI below.